postgres-appliance/bootstrap/pg_upgrade.py

```diff
@@ -28,6 +31,18 @@ def copy_configs(self):
                 shutil.copy(os.path.join(self._old_data_dir, f), os.path.join(self._data_dir, f))
         return True
 
+    def drop_possibly_incompatible_objects(self):
+        conn_kwargs = self._local_connect_kwargs
+        for p in ['connect_timeout', 'options']:
+            conn_kwargs.pop(p, None)
+
+        for d in self.query('SELECT datname FROM pg_catalog.pg_database WHERE datallowconn'):
+            logger.info('Executing "DROP SCHEMA IF EXISTS metric_helpers" in the database="%s"', d[0])
+            conn_kwargs['database'] = d[0]
+            with self._get_connection_cursor(**conn_kwargs) as cur:
+                cur.execute("SET synchronous_commit = 'local'")
+                cur.execute("DROP SCHEMA IF EXISTS metric_helpers CASCADE")
```

Comment:
Can we do this without dropping the schema?

Reply:
In theory it should be enough to drop function and view, but dropping the entire schema is the safest. post_init.sh will restore it after the upgrade.